### ISSUE TITLE
Object literal with async method in new expression's argument is not kept inline multi-lineable

### ISSUE THREAD

**Issue by dsherret** [dprint/dprint-plugin-typescript#23]
```ts
new ReadableStreamImpl({
    async pull(controller: ReadableStreamDefaultController): Promise<void> {
        try {
        const b = new Uint8Array(1024 * 32);
        const result = await read(fetchResponse.bodyRid, b);
        if (result === null) {
            controller.close();
            return close(fetchResponse.bodyRid);
        }

        controller.enqueue(b.subarray(0, result));
        } catch (e) {
        controller.error(e);
        controller.close();
        close(fetchResponse.bodyRid);
        }
    },
    cancel(): void {
        // When reader.cancel() is called
        close(fetchResponse.bodyRid);
    },
})
```

Formats incorrectly as:

```ts
new ReadableStreamImpl(
    {
        async pull(controller: ReadableStreamDefaultController): Promise<void> {
            try {
                const b = new Uint8Array(1024 * 32);
                const result = await read(fetchResponse.bodyRid, b);
                if (result === null) {
                    controller.close();
                    return close(fetchResponse.bodyRid);
                }

                controller.enqueue(b.subarray(0, result));
            } catch (e) {
                controller.error(e);
                controller.close();
                close(fetchResponse.bodyRid);
            }
        },
        cancel(): void {
            // When reader.cancel() is called
            close(fetchResponse.bodyRid);
        },
    },
);
```

**Comment by dsherret** [comment]
This is fixed once https://github.com/swc-project/swc/pull/877 is merged into swc. The problem is that the async keyword wasn't part of the method's span.
